rhessys/cycle/surface_daily_F.c

```diff
@@ -237,66 +270,105 @@ void		surface_daily_F(
 
 		/*** Calculate available energy at surface. Assumes Kdowns are partially ***/
 		/*** reflected by water surface based on water albedo. ***/
-		
-		if (zone[0].metv.dayl > ZERO) {
-				rnet_evap_pond = 1000 * ( (1-WATER_ALBEDO) * (patch[0].Kdown_direct + patch[0].Kdown_diffuse) 
-										+ patch[0].Lstar_pond
-										/* - patch[0].Lstar_soil */ /* Lstar_soil is assumed 0 when ponded water */
-										+ patch[0].surface_heat_flux ) 
-										/ zone[0].metv.dayl;
-				if (rnet_evap_pond <= ZERO) {
-					rnet_evap_pond = 0.0;
-				}
-			}
-			else rnet_evap_pond = 0.0;
-	
-			/*** Use Penman with rsurface=0 for open water evaporation. ***/
-	
-			patch[0].ga = max((patch[0].ga * patch[0].stability_correction),0.0001);
-		
-			detention_store_potential_dry_evaporation_rate = penman_monteith(
+
+//		rnet_evap_pond = 1000 * ( (1-WATER_ALBEDO) * (patch[0].Kdown_direct + patch[0].Kdown_diffuse)
+//				+ patch[0].Lstar_pond
+//				/* - patch[0].Lstar_soil */ /* Lstar_soil is assumed 0 when ponded water */
+//				+ patch[0].surface_heat_flux ) / zone[0].metv.dayl;
+		double rnet_evap_pond_night = 1000 * ( patch->Lstar_pond_night + surface_heat_flux_night) / nightlength;
+		double rnet_evap_pond_day = 1000 * ( (1 - WATER_ALBEDO) * (patch->Kdown_direct + patch->Kdown_diffuse)
+				+ patch->Lstar_pond_day + surface_heat_flux_day) / daylength;
+		rnet_evap_pond = rnet_evap_pond_night + rnet_evap_pond_day;
+		if (rnet_evap_pond <= ZERO) rnet_evap_pond = 0.0;
+		if (rnet_evap_pond_night <= ZERO) rnet_evap_pond_night = 0.0;
+		if (rnet_evap_pond_day <= ZERO) rnet_evap_pond_day = 0.0;
+
+		/*** Use Penman with rsurface=0 for open water evaporation. ***/
+
+		patch[0].ga = max((patch[0].ga * patch[0].stability_correction),0.0001);
+
+		detention_store_potential_dry_evaporation_rate_night = penman_monteith(
 				command_line[0].verbose_flag,
-				zone[0].metv.tday,
+				zone[0].metv.tnight,
 				zone[0].metv.pa,
-				zone[0].metv.vpd,
-				rnet_evap_pond,
+				zone[0].metv.vpd, // May need to calculate VPD for night
```

Comment:
yes I think we should use nightime vpd here - create new variable in zone - and do the calculation in zone daily F
and then we should also switch daytime vpd - and use this for ALL evaporation and transpiration calcs